### PR TITLE
Integration test for git archive without update

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
test/integration/targets/git/tasks/archive.yml

##### ANSIBLE VERSION
```
2.5-devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
cc @mkrizek

File: test/integration/targets/git/tasks/archive.yml [hunk 1]
```diff
@@ -32,3 +32,39 @@
   when:
     - "ansible_os_family == 'RedHat'"
     - ansible_distribution_major_version | version_compare('7', '>=')
+
+- name: ARCHIVE | Clear checkout_dir
+  file:
+    state: absent
```
Comment (mkrizek) [inline test/integration/targets/git/tasks/archive.yml]: If we clear the dir here then we're not testing https://github.com/ansible/ansible/pull/31829 because it will go through different code path, no? I guess instead of clearing the `checkout_dir` we need to clone it and then run the rest as is. Or am I missing something?

Comment (Akasurde) [reply test/integration/targets/git/tasks/archive.yml]: Yes. I am just clearing existing test environment and re-running git archive with update no. Does it make sense ?

Comment (mkrizek) [reply test/integration/targets/git/tasks/archive.yml]: Let me rephrase. To test https://github.com/ansible/ansible/pull/31829 we need following:
1. to have already cloned repo
2. run git module on that repo with `update` set to `no` and with `archive`

In the current patch we're missing 1. which means that we'll go here https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/git.py#L1072. But we need to go here https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/git.py#L1087 for which we need already cloned repo.

**Comment by resmo** [comment]
not sure about what state this PR is.

needs_info

File: test/integration/targets/git/tasks/archive.yml (hunk 1 above)
Comment (mkrizek) [reply test/integration/targets/git/tasks/archive.yml]: @Akasurde does that make sense?

Comment (Akasurde) [reply test/integration/targets/git/tasks/archive.yml]: OK. I will try again.

**Comment by Akasurde** [comment]
@mkrizek I updated patch, does it look the way you want ?

**Comment by mkrizek** [comment]
@Akasurde does my change look good to you? If so, feel free to merge.
